### DISCUSSION TITLE
ID providers sending EC-signed ID tokens not supported ?

### DISCUSSION THREAD

**Discussion by kwouters** [keycloak/keycloak#9624] [Q&A]
We're having some issues connecting an ID Provider that's running on Elliptic Curve crypto only.
When connecting it as an IDP in Keycloak, ID token verification (enabling "Validate Signatures" in the config) fails.

This is independent of "Use JWKS URL"; I tried both - with the JWKS URL and with the PEM formatted EC key listed explicitly in "Validating Public Key"

Question: is the verification of such tokens actually supported in Keycloak?
This might have something to do with it (but it's a guess, nothing more):

https://github.com/keycloak/keycloak/blob/8ed7c0544fb10e66e1e033b2c0f3b0fbdeeb6e28/services/src/main/java/org/keycloak/broker/oidc/OIDCIdentityProvider.java#L535

best,
Karel.

**Comment by andreasschmid-hypoport** [comment]
Exactly the same problem exists for me. 
Here is an example public JWKS URI using EC (elliptic curve) to better reproduce the problem: https://api.europace.de/auth/signing-keys

This is the error message that is provided in the Keycloak logs:
```
[...] ERROR [org.keycloak.broker.oidc.AbstractOAuth2IdentityProvider] (default task-497) Failed to make identity provider oauth callback: org.keycloak.broker.provider.IdentityBrokerException: token signature validation failed
at org.keycloak.keycloak-services@16.1.1//org.keycloak.broker.oidc.OIDCIdentityProvider.validateToken(OIDCIdentityProvider.java:557)
at org.keycloak.keycloak-services@16.1.1//org.keycloak.broker.oidc.OIDCIdentityProvider.validateToken(OIDCIdentityProvider.java:545)
```

Looking at the source code of `OIDCIdentityProvider`, line 557 throws this error because line 535 (quoted in Karel's post above) returns FALSE when calling `RSAProvider.verify` which makes sense, because no EC algorithms are found in class `RSAProvider`.

This is an important issue, because EC is used commonly.

Kind regards, Andreas

**Reply by andreasschmid-hypoport** [reply]
Created an issue for the problem as an enhancement request: https://github.com/keycloak/keycloak/issues/11036
